src/nodeFinders.ts

```diff
@@ -0,0 +1,53 @@
+import { SyntaxNode } from "web-tree-sitter";
+import { NodeFinder } from "./Types";
+
+export const findNode =
+  (isTargetNode: (node: SyntaxNode) => boolean): NodeFinder =>
+  (node: SyntaxNode) => {
+    return isTargetNode(node) ? node : null;
+  };
```

Comment:
has this been prettified?  I would think it would indent this one a bit differently.  I run prettier and eslint

Reply:
This is after prettier! This is how it formats an arrow function implicitly returning another arrow function. If we don't like the look of this I could always throw some curly braces and a return in there.

Reply:
ah right ok.  hmm.  could go either way.  i'm not super used to this style of function but people coming from FP languages prob dream in code like this.  fine with either